packages/actor-query-operation-quadpattern/lib/ActorQueryOperationQuadpattern.ts

```diff
@@ -49,12 +87,47 @@ export class ActorQueryOperationQuadpattern extends ActorQueryOperationTyped<Alg
 
   public async runOperation(pattern: Algebra.Pattern, context?: {[id: string]: any})
   : Promise<IActorQueryOperationOutputBindings> {
-    // Resolve the quad pattern
-    const result = await this.mediatorResolveQuadPattern.mediate({ pattern, context });
-
     // Collect all variables from the pattern
     const variables: string[] = this.getVariables(pattern);
 
+    let quadPattern: RDF.Quad = pattern;
+    let bindGraph: RDF.Term = null;
+    // Check if the available named graphs were specified
+    if (context && context.namedGraphs) {
```

Comment:
This does increase the interdependecy between the `quad` and `from` operators since all implementations for quad now have to know this exists unfortunately.

Reply:
Indirectly, yes. But as far as I can see, there is no clean way around this.
Also, theoretically, other actors _could_ add additional `namedGraphs` to the context. (That would be non-SPARQL though)

The alternative would be to do extensive query rewriting in the from actor, and I think this wouldn't be very efficient in some cases.

Reply:
My main worry is for when someone else wants to implement a new actor. If someone writes a new Quad actor they need to know `namedGraphs` is something that needs to be taken into account, otherwise it won't work nicely together with the existing FROM actor. And similarly if they write a FROM actor they again need to make sure they use that context field so this quad actor can work with it.

It's just something that we have to keep in mind and think about how to best provide this information to someone that wants to write a new actor. Just putting it in the JSDoc for this class specifically might not be sufficient since they might not think to look there.

Reply:
You're right, this is no good, will change it.